### PR TITLE
Fix `Paragraph.getOffsetForPosition` behavour in corner cases

### PR DESCRIPTION
**Pull request by mazunin-v-jb**
Follow-up to [the Skia/Skiko hit-testing fixes for caret positioning. ](https://github.com/JetBrains/skia/pull/20)
This PR must be merged after updating Skia.

- Enabled three previously ignored tests.
- Removed the workaround with the RTL block-position correction in `getOffsetForPosition` (the `correctedGlyphPosition -= 1` branch with the `isNeutralDirection` / `Direction.RTL` check). Skia should return the correct offset for clicks to the right of a line in mixed-direction text now, so the compose-side adjustment is no longer needed.
- Adjusted and removed getOffsetForPosition_insideComplexCharacter_shouldJumpToEnd` →
    `getOffsetForPosition_midpointOfComplexCharacter_snapsToClusterStart`, since it's more accurate and conforms to the Android behavior.

Fixes:
[CMP-8594 Fix `Paragraph.getOffsetForPosition` behavour in corner cases](https://youtrack.jetbrains.com/issue/CMP-8594)

## Testing
There are tests already, however, this better be tested by QA

## Release Notes
N/A

### CODE REVIEW

**Comment by mazunin-v-jb** [comment]
Just in case, it's better to merge it after [this PR](https://github.com/JetBrains/compose-multiplatform-core/pull/3049).

File: compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/SkiaParagraph.skiko.kt [hunk 1]
```diff
@@ -518,14 +518,6 @@ internal class SkiaParagraph(
             correctedGlyphPosition = paragraph.getGlyphPositionAtCoordinate(leftX + 1f, position.y).position
         } else if (position.x >= rightX) { // when clicked to the right of a text line
             correctedGlyphPosition = paragraph.getGlyphPositionAtCoordinate(rightX - 1f, position.y).position
-            val isNeutralChar = if (correctedGlyphPosition in text.indices) {
-                text.codePointAt(correctedGlyphPosition).isNeutralDirection()
-            } else false
-
-            // For RTL blocks, the position is still not correct, so we have to subtract 1 from the returned result
-            if (!isNeutralChar && getBoxBackwardByOffset(correctedGlyphPosition)?.direction == Direction.RTL) {
-                correctedGlyphPosition -= 1 // TODO Check if it should be CodePoint.charCount()
```
Comment (MatkovIvan) [inline compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/SkiaParagraph.skiko.kt:491]: It's an implementation of a common interface, and most likely some commonMain code relies on that.
Just saying that we do it differently and fixing copy-pasted tests (that should be in commonTest ideally) seems really suspicious tbh.

Shouldn't we align SkParagraph to android instead?

**Review by MatkovIvan — APPROVED** [review]
